platform/security/handler/security-handler-pki/src/main/java/org/codice/ddf/security/handler/pki/CrlChecker.java

```diff
@@ -0,0 +1,196 @@
+                                + " due to certificate being revoked by CRL.");
+                        LOGGER.warn(
+                                "Denying access for user {} due to certificate being revoked by CRL.",
+                                curCert.getSubjectDN());
+                        return false;
+                    }
+                }
+            } else {
+                String errorMsg = "The CRL check passed because CRL is disabled. Check that your properties and CRL file are correct if revocation is needed.";
+                SecurityLogger.logInfo(errorMsg);
+                LOGGER.warn(errorMsg);
+                return true;
+            }
+        } else {
+            LOGGER.debug(
+                    "Allowing message through CRL check. There were no certificates sent by the client.");
+            return true;
+        }
+        return true;
+    }
+
+    /**
+     * Sets the location of the CRL. Enables CRL checking if property is set, disables it otherwise
+     *
+     * @param location Location of the DER-encoded CRL file that should be used to
+     *                 check certificate revocation.
+     */
+    public void setCrlLocation(String location) {
+        if (location == null) {
+            LOGGER.warn("CRL property in [{}] is not set. Certs will not be checked against a CRL.",
+                    encryptionPropertiesLocation);
+            crl = null;
+            return;
+        }
+
+        crl = null;
+        try {
+            crl = createCRL(location);
+        } catch (FileNotFoundException fnfe) {
+            LOGGER.error("Could not find CRL file, cannot validate certificates to a CRL.", fnfe);
+            crl = null;
+        } catch (CertificateException ce) {
+            LOGGER.error(
+                    "Encountered an error while trying to create new certificate factory. Cannot validate certificates to a CRL.",
+                    ce);
+            crl = null;
+        } catch (CRLException ce) {
+            LOGGER.error(
+                    "Could not create new CRL from the provided file. File may not be valid CRL DER-encoded file.",
+                    ce);
+            crl = null;
+        }
+    }
+
+    /**
+     * Generates a new CRL object from the given location.
+     *
+     * @param location File location of the CRL file.
+     * @return new CRL object
+     * @throws FileNotFoundException If no file is located at the location
+     * @throws CertificateException  If the Certificate factory cannot be located
+     * @throws CRLException          If the input CRL file is invalid and cannot be used to
+     *                               generate a crl object.
+     */
+    private CRL createCRL(String location)
+            throws FileNotFoundException, CertificateException, CRLException {
+        try (FileInputStream fis = new FileInputStream(new File(location))) {
```

Comment:
Would be useful if you could abstract this so that the CRL file could also be a URL.